app/api/sessions.py

```diff
@@ -130,6 +130,17 @@ def before_get_object(self, view_kwargs):
             event = safe_query(self, Event, 'identifier', view_kwargs['event_identifier'], 'identifier')
             view_kwargs['event_id'] = event.id
 
+    def before_update_object(self, session, data, view_kwargs):
+        """
+        before update method to verify if session is locked before updating session object
+        :param event:
+        :param data:
+        :param view_kwargs:
+        :return:
+        """
+        if session.is_locked and data.get('is_locked') == session.is_locked:
+            raise ForbiddenException({'source': ''}, "Locked sessions cannot be edited")
```

Comment:
Point the source to the `isLocked` property like it is done for marshmallow validations.

Reply:
@CosmicCoder96 I've updated it. Please review